x/ibc-rate-limit/types/params.go

```diff
@@ -0,0 +1,65 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+// Parameter store keys.
+var (
+	KeyContractAddress = []byte("contract")
+)
+
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+func NewParams(contractAddress string) (Params, error) {
+	return Params{
+		ContractAddress: contractAddress,
+	}, nil
+}
+
+// default gamm module parameters.
+func DefaultParams() Params {
+	return Params{
+		ContractAddress: "",
+	}
+}
+
+// validate params.
+func (p Params) Validate() error {
+	if err := validateContractAddress(p.ContractAddress); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Implements params.ParamSet.
+func (p Params) ParamSetPairs() paramtypes.ParamSetPairs {
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(KeyContractAddress, p, validateContractAddress),
+	}
+}
+
+func validateContractAddress(i interface{}) error {
```

Comment:
None of the other of the `validate` methods on params I have seen in other modules have tests. Is there a reason for having them specifically here?

Reply:
In my opinion, It isn't great that the some of the modules are missing these tests. We've run into issues numerous times where some value in parameters or genesis state would be invalid during init / export genesis, causing hard-to-debug problems on testnets. So covering these with tests would be helpful.

It seems that for some of the newer modules, we have been trying to maximize test coverage even for params validation:
https://github.com/osmosis-labs/osmosis/blob/c676404b6569f7d4547761a19fb7539973facf1f/x/twap/types/export_test.go#L35

Reply:
@p0mvn got any idea why the tests (https://github.com/osmosis-labs/osmosis/blob/20b40deacc9143cc0b9393def410e8b84e4bd759/x/ibc-rate-limit/types/params_test.go) would expect the cosmos bech32 prefix instead of osmo? I still think it makes sense to have the test, but it's driving me crazy that the setup is wrong